examples/rename_brainvision_files.py

```diff
@@ -0,0 +1,98 @@
+"""
+=================================
+Rename BrainVision EEG data files
+=================================
+
+The BrainVision file format is one of the recommended formats to store EEG data
+within a BIDS directory. To organize EEG data in BIDS format, it is often
+necessary to rename the files. In the case of BrainVision files, we would have
+to rename multiple files for each dataset instance (i.e., once per recording):
+
+1. A text header file (``.vhdr``) containing meta data
+2. A text marker file (``.vmrk``) containing information about events in the
+data
+3. A binary data file (``.eeg``) containing the voltage values of the EEG
+
+The problem is that the three files contain internal links that guide a
+potential data reading software. If we just rename the three files without also
+adjusting the internal links, we corrupt the file format.
+
+In this example, we use MNE-BIDS to rename BrainVision data files including a
+repair of the internal file pointers.
+"""
+
+# Authors: Stefan Appelhoff <stefan.appelhoff@mailbox.org>
+# License: BSD (3-clause)
+
+###############################################################################
+# We are importing everything we need for this example:
+import os
+import requests
+
+from mne_bids.utils import copyfile_brainvision
+from mne.io import read_raw_brainvision
+
+###############################################################################
+# Step 1: Download some example data
+# ----------------------------------
+# To demonstrate the MNE-BIDS functions, we need some testing data. Here, we
+# will use data that is shipped with MNE-Python.
+
+# Make a directory to save the data to
+home = os.path.expanduser('~')
+examples_dir = os.path.join(home, 'mne_data', 'mne_bids_examples')
+if not os.path.exists(examples_dir):
+    os.makedirs(examples_dir)
+
+# Use the requests library to get content from github and save it
+base_url = 'https://github.com/mne-tools/mne-python/'
+base_url += 'raw/master/mne/io/brainvision/tests/data/test'
+file_endings = ['.vhdr', '.vmrk', '.eeg', ]
+
+for f_ending in file_endings:
+    url = base_url + f_ending
+    response = requests.get(url)
+
+    fname = os.path.join(examples_dir, 'test' + f_ending)
+    with open(fname, 'wb') as fout:
+        fout.write(response.content)
```

Comment:
This should be made a function. The downloading of data should be hidden from user.

Reply:
Why should the downloading be hidden from the user? I doubt that making a function for this will be useful, because this data is unlikely to be reused in other examples or code snippets. However as soon as we get a second usecase, I'd be happy to refactor and adjust ...

Reply:
The reason is that you want to hide complexity :-) Expose the user to your tool, not complicated Python ...

Reply:
broader question: is it possible to download specific files with the mne instead of the whole dataset?

I am actually fine with having the download steps here. maybe we could include in the heading of the section. replace this section with your local data if you have it or something like this.

Reply:
People stop following the moment they don't understand something. I really don't like the idea of making a 75 line example when you can do it in 3 lines ...

Reply:
okay I got all the numbers wrong there :) But you get the idea ... 

it's also the philosophy behind `mne` and `sklearn` too. The fetchers don't exist just because code needs to be DRY.

Reply:
Okay - so you'd be okay with me making a `_get_brainvision_test_data` function in `utils.py`?

Reply:
umm ... we have `mne_bids/datasets.py`. We could add it there and it would be a public function.